src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(health, name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name, hidden) {
+    super(health, name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(enemy) {
+    if (!enemy.hidden && enemy.constructor.name === 'Herbivore') {
+      enemy.health -= 50;
+    }
+
+    if (enemy.health === 0 && Animal.alive.indexOf(enemy) !== -1) {
+      Animal.alive = Animal.alive.filter(beast => beast !== enemy);
+    }
```

Comment:
you don't need to use `indexOf` here

Reply:
@mykola-koval it was stated in the checklist that we should check if rabbit is herbivore